### ISSUE TITLE
Could not register Flint ChromeOS as a Digital Signage device

### ISSUE THREAD

**Issue by dagwieers** [flintinnovations/overlay-rpi#1]
So we where hoping for an elementary school to use FlintOS as a cheap general-purpose ChromeOS device as a Digital Signage device: https://enterprise.google.com/chrome/digital-signage/

However, when attempting to register the device it fails because the device-type is unknown. Do you know if this is at all possible, or whether there's a possibility to work together with Google to get devices like FlintOS supported ?

**Comment by Simple-Will** [comment]
Hey, 

It is possible we have got it working internally by editing the machine-info to pull the device serial number but it's not the most elegant solution. We're looking to implement something more robust once the freon builds a more stable particularly booting via uboot so we can move to 4.12 kernel.

**Comment by dagwieers** [comment]
That's good to hear. Keep us informed, I don't mind helping test where needed.

**Comment by dagwieers** [comment]
We've come at a point that I either need this to work, or I will have to buy a chromebox device. My preference would be to help out with this project rather than succumb to a proprietary solution.

**Comment by Simple-Will** [comment]
Hey, sorry we are focusing our dev time on the PC builds at the moment, we will add in the capability to enterprise enroll the pi to the git hub hopefully today/tomorrow but it will only really be usable on the now quite outdated x11 builds which run on r56 of chromium OS. 

Hopefully we will find the time to look at stabilising the freon builds soon or the community starts to make some headway!

**Comment by dagwieers** [comment]
Thanks for the info. I understand the need for a focus and the limited resources, but if there's anything you can share here on how to do this, I don't mind looking into this myself.

**Comment by giriflint** [comment]
```#!/bin/sh

# Copyright 2016 The Chromium OS Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

set -e

if [ -e /var/lib/enterprise_serial_number_recovery -o \
     ! -e /var/lib/whitelist/owner.key ]; then
  touch $MACHINE_INFO
  chmod 0644 $MACHINE_INFO
  if crossystem "mainfw_type?nonchrome"; then
    echo "serial_number=\"nonchrome-$(date -u +%s)\""
  else
    # Dump full information in the VPD, including the serial number.
    #dump_vpd_log --full --stdout
    echo "serial_number=\"`cat /proc/cpuinfo | tail -n 3 | cut -d ' ' -f 2 | head -n 1`-`cat /proc/cpuinfo | tail -n 3 | cut -d ' ' -f 2 | tail -n 1`\""
  fi >> $MACHINE_INFO
fi
```
This is the write-machine-conf that should go into /usr/share/cros/init.
You would have to remount your fs to r/w with `sudo mount -o remount, rw` and replace the contents of the existing write-machine-conf with the above.

Ideally, you should write an ebuild that copies this file into /usr/share/cros/init. Please check gentoo wiki for details on how to do that. It should be simple. You can contact us in case you need any help. Once you have the ebuild ready, you can submit a merge request (pull request).

**Comment by dagwieers** [comment]
I have verified and this seems to work well.
We seem to have an issue with device enrollment on G Suite for Education.
It tells me we _"have insufficient licenses for enrolling this device"_.

Looking at this today more closely, using **awk** is a bit more reliable.
(i.e. not relying on position positions in a file)

So my solution is this:

```bash
# Copyright 2016 The Chromium OS Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

set -e

if [ -e /var/lib/enterprise_serial_number_recovery -o \
     ! -e /var/lib/whitelist/owner.key ]; then
  touch $MACHINE_INFO
  chmod 0644 $MACHINE_INFO
  if crossystem "mainfw_type?nonchrome"; then
    echo "serial_number=\"nonchrome-$(date -u +%s)\""
  else
    # Dump full information in the VPD, including the serial number.
    #dump_vpd_log --full --stdout
    awk '/^Hardware\s/{hw=$3} /^Serial\s/{nr=$3} {next} END {print "serial_number=" hw "-" nr}' < /proc/cpuinfo
  fi >> $MACHINE_INFO
fi
```

**Comment by giriflint** [comment]
So, does using this approach solve your ¨_insufficient licenses_¨ problem ?

**Comment by dagwieers** [comment]
No, I think the licensing issue is related to the fact that we are moving from a Legacy Google Apps to a trial G Suite for Education. Before it complained about the licenses, it complained about the device type/serial, so that's why I am confident this works now.

It may take about 2 weeks before our G Suite for Education is approved. As soon as I have something working completely, I will reporting back.

**Comment by dagwieers** [comment]
Alright, we now have an official G Suite for Education, and I tried again to enroll my device, and I still am getting the message that I have insufficient licenses for registering this device.  So this error is unrelated to the fact we had a trial version of G Suite.

Looking at the documentation at: https://support.google.com/chrome/a/answer/1360534 it seems it is expected you purchase licenses in order to enroll devices to G Suite. I have to discuss with Google Support regarding if this is true, as it would make Chromebox devices in general a less affordable solution for e.g. digital signage.

**Comment by Simple-Will** [comment]
Hi Dag

Yes you will need chrome device licences inorder to enroll the device.
There are single use kiosk licences available. We don't currently sell
these as it's quite a long process to get approval to sell them. I'm not
sure of your setup but if you used a 3rd party to help with the
implementation of G suite for education they will be able to help you get a
relevant kiosk licence. You can also sign up for a trial for 60 days, this
can be done through the Google admin settings.

On 10 Nov 2017 12:32 p.m., "Dag Wieers" <notifications@github.com> wrote:

Alright, we now have an official G Suite for Education, and I tried again
to enroll my device, and I still am getting the message that I have
insufficient licenses for registering this device. So this error is
unrelated to the fact we had a trial version of G Suite.

Looking at the documentation at: https://support.google.com/chr
ome/a/answer/1360534 it seems it is expected you purchase licenses in order
to enroll devices to G Suite. I have to discuss with Google Support
regarding if this is true, as it would make Chromebox devices in general a
less affordable solution for e.g. digital signage.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/flintinnovations/overlay-rpi/issues/1#issuecomment-343536762>,
or mute the thread

**Comment by dagwieers** [comment]
Good news, I was able to register it with a Chrome device license. I configured the device through the G Suite management tool, created a schedule using Google Sign Builder and configured it to autostart. However none of the settings to the org unit of the device seem to impact the device. (e.g. I disable guest login, but it is still there on the device).

So I don't know whether this is a problem with Flint, with Google Sign Builder (or ChromeOS in itself) or a user error. I suspect the latter, but I have no clue...

**Comment by giriflint** [comment]
Do you need to reboot after changing those settings? Can you share the exact list of steps that you tried. It could most likely be an issue with the procedure related to managing an enrolled device.

**Comment by covex33** [comment]
Can I get info on where to put the code? or at least a link to what you are talking about? I am having the same issue here and would like a little more instruction for a non programmer.

**Comment by RedXTaz** [comment]
Is it working with the new FydeOS on raPi 4? I'm not able to enroll this device.
